spec/ddtrace/configuration_spec.rb

```diff
@@ -225,6 +226,8 @@
             test_class.configure do |c|
               c.runtime_metrics.statsd = new_statsd
             end
+
+            expect(old_statsd).to_not have_received(:close)
```

Comment:
I've finally found a case where `expect(...).to receive(...)` is not equivalent to `expect(...).to have_received(...)`.

In this specific case, `old_statsd.close` is invoked in an `after` block, which trips `expect(...).to receive(...)`, but not `expect(...).to have_received(...)`.

Reply:
Good point, I had never considered that the `have_received` has a temporal component, it's "this thing has/hasn't happened so far -- doesn't say anything about the future". (But to be honest I never use it so maybe that's why I hadn't run into this before :P)

Although -- may I suggest moving the expectation to the `it`? It's a bit weird to have it in the before, since the it specifically states "...but does not close the old Statsd".